pkg/util/version/stream_test.go

```diff
@@ -41,40 +41,58 @@ func TestGetStream(t *testing.T) {
 		Version: NewVersion(4, 4, 3),
 	}
 
-	Streams = append([]Stream{}, Stream43, Stream44)
 	for _, tt := range []struct {
-		name string
-		v    *Version
-		want Stream
-		err  error
+		name    string
+		v       *Version
+		streams []Stream
+		want    Stream
+		err     error
 	}{
 		{
-			name: "4.3 - upgrade",
-			v:    NewVersion(4, 3, 17),
-			want: Stream43,
+			name:    "single stream - upgrade x.Y",
+			v:       NewVersion(4, 3, 9),
+			streams: []Stream{Stream44},
+			want:    Stream44,
 		},
 		{
-			name: "4.3 - error",
-			v:    NewVersion(4, 3, 19),
-			err:  fmt.Errorf("not upgrading: cvo desired version is 4.3.19"),
+			name:    "single stream - upgrade X",
+			v:       NewVersion(4, 4, 11),
+			streams: []Stream{Stream44},
+			want:    Stream44,
 		},
 		{
-			name: "4.4 - upgrade",
-			v:    NewVersion(4, 4, 2),
-			want: Stream44,
+			name:    "multi stream - upgrade x.Y lower",
+			v:       NewVersion(4, 3, 17),
+			streams: []Stream{Stream43, Stream44},
+			want:    Stream43,
 		},
 		{
-			name: "4.4.10 stream",
-			v:    NewVersion(4, 4, 9),
-			err:  fmt.Errorf("not upgrading: cvo desired version is 4.4.9"),
+			name:    "multi stream - upgrade x.Y higher",
+			v:       NewVersion(4, 3, 19),
+			streams: []Stream{Stream43, Stream44},
+			want:    Stream43,
 		},
 		{
-			name: "error",
-			v:    NewVersion(4, 5, 1),
-			err:  fmt.Errorf("stream for 4.5.1 not found"),
```

Comment:
I see this test case is removed... Our test cases aren't checking if we're protected against downgrades between minor versions anymore. Please add a case to check a target version of `4.5.x` to ensure we don't have a regression.

Reply:
This moved places into upgrade_test.go as I think this was the wrong place in the first place.